cmd/operator-sdk/generate/csv.go

```diff
@@ -53,35 +55,58 @@ for the operator. This file is used to publish the operator to the OLM Catalog.
 
 A CSV semantic version is supplied via the --csv-version flag. If your operator
 has already generated a CSV manifest you want to use as a base, supply its
-version to --from-version. Otherwise the SDK will scaffold a new CSV manifest.
+version to --from-version. Otherwise the SDK will scaffold a new CSV manifest.`,
+		Example: `
+TODO
+	`,
 
-Configure CSV generation by writing a config file 'deploy/olm-catalog/csv-config.yaml`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			// The CSV generator assumes that the deploy and pkg directories are
-			// present at runtime, so this command must be run in a project's root.
-			projutil.MustInProjectRoot()
-
 			if len(args) != 0 {
 				return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
 			}
 			if err := c.validate(); err != nil {
 				return fmt.Errorf("error validating command flags: %v", err)
 			}
+			// Default for crd dir if unset
+			if c.crdDir == "" {
+				c.crdDir = c.deployDir
```

Comment:
It would still be necessary to change the signature.
Users can set `--crd-dir` to whatever they want.
So if they set it to `--crd-dir=deploy` or some other directory which has the `olm-catalog` as a subdirectory then `GetCRDManifestPaths()` needs to ignore that in the filepath Walk.

The signature change is mostly because the `GetCRDManifestPaths()` util is shared between `generate crd` where the crd directory is always fixed to `deploy/crd` and `generate csv --crd-dir` where the CRD directory could be any path.
I could split this into two separate almost identical utils for each command but this is a smaller change.

Also `--crd-dir=deploy` or `c.crdDir = c.deployDir` because that's essentially what it was before this refactor when there was no `crd-dir` flag. Let me know if you think we should change that default though, unrelated to the `GetCRDManifestPaths()` signature change.

Reply:
In that case, we should:

1. Change `GetCRDManifestPaths` so that it does a shallow (depth 1) search for CRDs. This is more intuitive IMO, since there's nothing in the name that says it performs a recursive search. Now when users pass a dir without CRDs in it (ex. `deploy`) they should see an error that no CRDs were present in that dir.
1. Pass `crdDir` to `Config.Inputs` for the CRD generator, which should be looking in that directory anyway.
1. Default `crdDir` to `filepath.Join(c.deployDir, "crds")`.

Where is `GetCRDManifestPaths` used by `generate crds`?

Reply:
@estroz `GetCRDManifestPaths()` is used by `generate crds` here:
https://github.com/operator-framework/operator-sdk/blob/master/internal/util/k8sutil/crd.go#L32

If we're doing a shallow read instead I'll leave that util untouched and just do a simple read in `generate csv`.

Reply:
Remove this default since it is now set during flag registration.